File: otherSolution/js/week2.js
const getLongestAscendingSequence = (data) => {
    let longest = 1
    let longestArr = []
    for (let i = 0; i + 1 < data.length; i++) {
        if (data[i] <= data[i + 1]) {
            longest++;
        } else {
            longestArr.push(longest)
            longest = 1;
        }
    }
    longest = Math.max(...longestArr);
    let sequence = []
    for (let i = 0; i + 1 < data.length; i++) {
        if (data[i] <= data[i + 1] ) {
            sequence.push(data[i])
            sequence.push(data[i + 1])
        }
        else if (data[i] <= data[i + 1] && i + 1 === data.length - 1) {
           
        } else {
            sequence = []
        }

        if(sequence.length === longest){
            return sequence
        }
        console.log(sequence)
    }
    return sequence
}

console.log(getLongestAscendingSequence([70, 70, 20, 10, 80, 20]))

File: otherSolution/python/week2.py
#!/usr/bin/env python3

def intersection(a: list, b: list):
    result = 0
    for i in range(len(a)):
        for j in range(len(b)):
            if a[i] == b[j]:
                result += 1
                del b[j]
                break
    return result


def getLongestAscendingSequence(data):
    info = {}
    numbers = []
    for num in data:
        if len(numbers) == 0 or num >= numbers[-1]:
            numbers.append(num)
        else:
            if len(numbers) not in info:
                info[len(numbers)] = numbers
            numbers = [num]
    info[len(numbers)] = numbers
    return info[max(info, key=(lambda k: info[k] and k))]


def getLongestUnchangedSequence(data):
    info = {}
    numbers = []
    for num in data:
        if len(numbers) == 0 or num == numbers[-1]:
            numbers.append(num)
        else:
            if len(numbers) not in info:
                info[len(numbers)] = numbers
            numbers = [num]
    if len(numbers) not in info:
        info[len(numbers)] = numbers
    return info[max(info, key=(lambda k: info[k] and k))]

def checkIfSorted(data,ascending):
    if ascending:
        if(all(data[i] > data[i + 1] for i in range(len(data) - 1))):
            return False
    else:
        if(all(data[i] < data[i + 1] for i in range(len(data) - 1))):
            return False
    return True

def sortIfNeeded(data,asc):
    ascending = True if data[0] < data[-1] else False
    if checkIfSorted(data,ascending):
        return
    if asc:
        data = sorted(data)
    else:
        data = sorted(data,reverse=True)

